source/tests/e2e/test_app.py

```diff
@@ -0,0 +1,61 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+from webdriver_manager.chrome import ChromeDriverManager
+from selenium import webdriver
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.chrome.options import Options
+from selenium.webdriver.support.ui import Select
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.common.by import By
+
+@pytest.fixture
+def browser():
+    chrome_options = Options()
+    ####### TESTING - remove headless to see browser actions
+    # Make sure the window is large enough in headless mode so that all
+    # the elements on the page are visible
+    chrome_options.add_argument("--headless")
+    chrome_options.add_argument("--window-size=1920,1080")
+    chrome_options.add_argument("--start-maximized")
+    ####### TESTING - remove headless to see browser actions
+    from selenium import webdriver
+
+    browser = webdriver.Chrome(ChromeDriverManager().install(), chrome_options=chrome_options)
+    return browser
+
+
+def test_everything(browser, test_environment, stack_resources):
+    browser.implicitly_wait(5)
+    browser.get(stack_resources['UserInterface'])
+    wait = WebDriverWait(browser, 30)
+    ####### Login
+    username_field = browser.find_element("xpath", "/html/body/div/div/div/div/div[2]/div[1]/div/input")
```

Comment:
np: `/html/body/div/div/div/div/div[2]/div[1]/div/input` will this be a problem if another 30 child `div`'s are added in the future.

Reply:
The login page doesn't change very often, but yes, this xpath is tightly coupled to the current DOM. If the Amplify team changes their login page, then we'll need to update the xpath.